hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -94,38 +107,13 @@ public boolean shouldThrottleQuery(HederaFunctionality queryFunction, Query quer
 
 	@Override
 	public boolean shouldThrottleTxn(final TxnAccessor accessor, final Instant now) {
+		resetLastAllowedUse();
```

Comment:
This call and on line #113 appear to change the contract of the method, as well as possibly its performance profile... 🤔 It would be **very** nice if we can only extend/compose `DeterministicThrottling` to get what we need for scheduled throttling. (The current implementation have been extensively tested to enforce the expected TPS in real networks, and if at all possible we shouldn't risk breaking it!)

Reply:
First - this actually fixes a bug - gas will drain and not be reclaimed if we throttle based on the txn type.

Also, we need something like this to reclaim the ScheduleCreate throttles if the contained txn causes it to throttle.